### ISSUE TITLE
Reconsider mailing list host

### ISSUE THREAD

**Issue by gdt** [bcpierce00/unison#711]
The current list can't be administered without a google account, and I've received reports of vanishing (non-spammy and entirely appropriate) messages.   I'm unclear on possibilities and this seems like a lot of work, but it should be on the eventual list.

**Comment by bcpierce00** [comment]
Are any of the vanishing messages recent?

There was a bug in my email client that caused some submitted messages to
display as blank.  I thought these were spam, but realized a couple months
ago that if I went to the actual list web site I could see them.

**Comment by gdt** [comment]
I got an offlist email about a submission gone missing.  I'lll followup with you off-github about that.

I have also been seeing complaints about missing messages on other googlegroups lists.

**Comment by gdt** [comment]
Also, the current list mangles URLs.

**Comment by gdt** [comment]
Also, there was a report just now on unison-users@ which says that the mechanism to join without a google account does not work.  If that's really true, that's a serious problem.

**Comment by bcpierce00** [comment]
Agreed.

One other issue that I'm not sure about is how easy it is to export the
list of current subscribers from a Google group -- i.e., whether we'd be
able to easily move everybody over to another platform.  But if people want
to move somewhere else and can agree on where and how, I'm happy to go
along.

**Comment by gdt** [comment]
One possible home is https://www.freelists.org/

**Comment by llucps** [comment]
I'm guessing that this a low priority task, just wondering whether freelists.org is an option a the end.

**Comment by gdt** [comment]
As I see it, requirements are:
  - no charges
  - no requirement for list admins to agree to indemnification or other unreasonable terms
  - no requirement for list members to have a google account or to agree to unreasonable terms
  -  ability to have a list with 1000 or so members (so a scheme that allows < 5000 is ok)
  - non-members are blocked
  - decent inbound spam filtering
  - proper X-List headers with reasonable unsubscribe ability
  - list configurable and configured to not break DKIM (no subject tag, no footer)
  - list applies DKIM signature from the list
  - well-run server not in RBLs
  - public archives
non-requirements:
  - digest mode (perhaps "not having digests" is a requirement; that leads to horrific quoting)
  - forum interface

**Comment by gdt** [comment]
> I'm guessing that this a low priority task, just wondering whether freelists.org is an option a the end.
> 
> I just try the options listed here https://webapps.stackexchange.com/questions/13508/how-can-i-subscribe-to-a-google-mailing-list-with-a-non-google-e-mail-address
> 
> But sadly none of them worked for me.

As of not that long ago, the method at https://github.com/bcpierce00/unison/wiki/Mailing-Lists
of sending to `unison-users+subscribe@seas.upenn.edu` worked.   What happened when you tried that?

**Comment by llucps** [comment]
> > I'm guessing that this a low priority task, just wondering whether freelists.org is an option a the end.
> > I just try the options listed here https://webapps.stackexchange.com/questions/13508/how-can-i-subscribe-to-a-google-mailing-list-with-a-non-google-e-mail-address
> > But sadly none of them worked for me.
> 
> As of not that long ago, the method at https://github.com/bcpierce00/unison/wiki/Mailing-Lists of sending to `unison-users+subscribe@seas.upenn.edu` worked. What happened when you tried that?
> 
> I just sent a request, and got
> 
> ```
> We have received a request to join the group unison-users@seas.upenn.edu.
> You are already subscribed to this group.
> ```

I get a Delivery Status notification (delivery) email back with the following message:

![Screenshot 2024-05-01 at 18 55 49](https://github.com/bcpierce00/unison/assets/25823659/67ee0b1b-e368-4d6b-a558-a166119782a8)

**Comment by gdt** [comment]
In general I only read text not images, but I was able to reproduce this.   I've sent a query and we'll see.

**Comment by llucps** [comment]
> In general I only read text not images, but I was able to reproduce this. I've sent a query and we'll see.

Oh sorry the error is:

```
Message not delivered

There was a problem delivering your message to unison-users@gmailrelay.seas.upenn.edu. See the technical details below.
The response was:

Unexpected DFS AddMemberships membership status 
```

**Comment by bcpierce00** [comment]
I can check with my sysadmins, but just to make sure I understand clearly: This was in response to a message sent to unison-users@seas.upenn.edu?  Would you mind sharing the address from which it was sent?  (You can send it to me privately if you prefer.)

**Comment by llucps** [comment]
> I can check with my sysadmins, but just to make sure I understand clearly: This was in response to a message sent to [unison-users@seas.upenn.edu](mailto:unison-users@seas.upenn.edu)? Would you mind sharing the address from which it was sent? (You can send it to me privately if you prefer.)

Yes that was an email sent from my personal email to subscribe to unison-users@seas.upenn.edu mailing list.

I'll send you my email by private message.

Thanks

**Comment by gdt** [comment]
We are not talking about messages sent to the ilst.  We are talking about messages sent to `unisonusers+subscribe@seas.upenn.edu` so that someone can join the list.

To be clear, the problem is not that any particular method doesn't work.  It is that we lack a documented, working method for someone (who does not have a google account) to join the list (without telling them to get one!).

**Comment by llucps** [comment]
> We are not talking about messages sent to the ilst. We are talking about messages sent to `unisonusers+subscribe@seas.upenn.edu` so that someone can join the list.
> 
> To be clear, the problem is not that any particular method doesn't work. It is that we lack a documented, working method for someone (who does not have a google account) to join the list (without telling them to get one!).

Yes you're right, I should've been more specific.

**Comment by bcpierce00** [comment]
Now I'm not sure I understand.  I thought the problem was that we _do_ have a documented method but that this particular method is not working... or?

**Comment by gdt** [comment]
This is the documented method.  It didn't work for @llucps who got a bizarre bounce.   I was able to replicate this by creating a new email address and sending a request from it, getting the same bizarre bounce (with change of value of sending address).   That's what I sent to you via email.

You should be able to replicate this by sending a request from an email address that is not on the list.   Or if you can't replicate it, that's interesting too.   (It would not shock me if upenn addrs are treated differently.)

**Comment by bcpierce00** [comment]
Also, just double checking: the subscription address is
***@***.***
***@***.***>, not
***@***.***

Is the former the one you used?

**Comment by bcpierce00** [comment]
Oof -- that did not come through clearly.  What I meant to write was that the correct address is unison-users+subscribe, not unisonusers+subscribe.

**Comment by gdt** [comment]
yes, I had the hyphen.   I messed it up when copying to comment.  You can see this clearly in the bounce I forwarded to you.

**Comment by justinlovinger** [comment]
Although not technically a mailing list, using GitHub Discussions is an option maybe worth considering. It would integrate better with GitHub Issues. When someone opens an issue better suited as a discussion, it can be seamlessly converted to a discussion with breadcrumbs for future searches of similar issues.
